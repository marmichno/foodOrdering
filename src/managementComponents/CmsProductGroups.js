import {CmsNavbar} from './CmsNavbar';
import {useEffect, useState} from 'react';
import {groupPostRequest} from '../requests/groupPostRequest';
import {groupGetRequest} from '../requests/groupGetRequest';
import {groupDeleteRequest} from '../requests/groupDeleteRequest';

export const CmsProductGroups = () =>{

    const [groups, setGroups] = useState([]);
    const [inputValue, setInputValue] = useState("");

    useEffect(() => {
        getGroups();
    },[]);

    const getGroups = async () =>{
        const response = await groupGetRequest();
        setGroups(response);
    }

    const addGroup = async () =>{
        const response = await groupPostRequest(inputValue);
        getGroups();
    }

    const deleteGroup = async (e) =>{
        const response = await groupDeleteRequest(e.target.dataset.id);
        getGroups();
    }

    return(
        <div className="productGroupsMainContainer">
            <CmsNavbar/>
            <div className="productGroupsContent">
                <div className="groupRow">
                    <div><h1>id</h1></div>
                    <div><h1>Group name</h1></div>
                    <div><h1>Products in group</h1></div>
                    <div><h1>Edit</h1></div>
                </div>

                <div className="addGroupContainer">
                    <input onChange={(e) => setInputValue(e.target.value)} type="text" placeholder="group name" autocomplete="off" name="category" required></input>
                    <button onClick={addGroup}>Add product group</button>
                </div>

                {
                groups !== undefined ?
                    groups.map(value => {
                        return(
                            <div className="groupRow">
                                <div>{value.id}</div>
                                <div><p>{value.name}</p></div>
                                <div><p>26</p></div>
                                <div><button>modify</button><button data-id={value.id} onClick={deleteGroup}>delete</button></div>
                            </div>
                        )
                    })
                : null
                }

            </div>
        </div>
    )
}