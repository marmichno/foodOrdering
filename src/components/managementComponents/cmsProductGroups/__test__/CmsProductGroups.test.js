import {CmsProductGroups} from '../CmsProductGroups';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import mockedAxios from 'axios';
import validGroupsJson from '../__mocks__/validGroups.json';
import TestRenderer from "react-test-renderer";
import {render, fireEvent, waitFor, screen} from '@testing-library/react';

const store = createStore(() => [], {}, applyMiddleware());

test("checks if groups are visible after api call", async () => {
    mockedAxios.get.mockResolvedValueOnce({data: validGroupsJson});

    const {getAllByText} = render(<Provider store={store}><CmsProductGroups/></Provider>);

    await waitFor(() => {
        expect(getAllByText('sushi rolls'));
        expect(getAllByText('prawns'));
        expect(getAllByText('others'));
        expect(getAllByText('sets'));
    })
});

test("matches snapshot", () => {
    const tree = TestRenderer.create(<Provider store={store}><CmsProductGroups/></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});
