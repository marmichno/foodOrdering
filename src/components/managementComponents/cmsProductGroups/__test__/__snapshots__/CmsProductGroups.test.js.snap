// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="productGroupsMainContainer"
>
  <div
    className="hamburgerMenu"
  >
    <svg
      fill="none"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="3"
        x2="21"
        y1="12"
        y2="12"
      />
      <line
        x1="3"
        x2="21"
        y1="6"
        y2="6"
      />
      <line
        x1="3"
        x2="21"
        y1="18"
        y2="18"
      />
    </svg>
  </div>
  <div
    className="productGroupsContent"
  >
    <div
      className="groupRow"
    />
    <div
      className="groupRow"
    >
      <div>
        <h1>
          id
        </h1>
      </div>
      <div>
        <h1>
          Group name
        </h1>
      </div>
      <div>
        <h1>
          Products in group
        </h1>
      </div>
      <div>
        <h1>
          Edit
        </h1>
      </div>
    </div>
    <div
      className="addGroupContainer"
    >
      <input
        autocomplete="off"
        name="category"
        onChange={[Function]}
        placeholder="group name"
        required={true}
        type="text"
      />
      <button
        onClick={[Function]}
      >
        Add product group
      </button>
    </div>
  </div>
</div>
`;
